### PR TITLE
dev: Develop Order Service.

### PR DESCRIPTION
**Pull request by kai890707**
**Description**

1.  Implement Order service logic.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] Conforms to style guide

### CODE REVIEW

**Review by ping-yee — CHANGES_REQUESTED** [review]
There is something need to be changed.

File: develop/app/Controllers/v2/OrderController.php [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+        if (is_null($p_key)|| is_null($amount)) {
+            return $this->fail("Incoming data error", 404);
+        }
+
+        $orderModel   = new OrderModel();
+
+        $now        = date("Y-m-d H:i:s");
+        $orderData  = [
+            "o_key"        => sha1($u_key . $p_key . $now . random_int(0, 10000000)),
```
[the remark this answers is not in the document]
Comment (kai890707) [reply develop/app/Controllers/v2/OrderController.php:131]: Do you mean that the o_key does not need to be created by the service, but the o_key attached to the data is directly used as input when the data is passed in?

[the remark this answers is not in the document]
Comment (ping-yee) [reply develop/app/Controllers/v2/OrderController.php:131]: I think that there are two ways to implement order key with idempotence:
1. Creating the order key by hash the order information(like: now time, products data and user key... et al.) when developer define the orchestation steps and pass in to the order service.
2. Only using the hashing order info to generate order key in order service, in other words, don't use that random int and judge whether the generated order key already exists in the database. If it is, respond the error and make the user try it again.

[the remark this answers is not in the document]
Comment (kai890707) [reply develop/app/Controllers/v2/OrderController.php]: This change is done.

**Review by ping-yee — APPROVED** [review]
LGTM!

**Pull request merged**